google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -107,9 +106,6 @@ public Void apply(WriteLogEntriesResponse input) {
 
   private static <I, O> Future<O> transform(Future<I> future,
       Function<? super I, ? extends O> function) {
-    if (future instanceof ListenableFuture) {
-      return Futures.transform((ListenableFuture<I>) future, function);
-    }
```

Comment:
It has never been necessary. `Futures.transform` and `Futures.lazyTransform` do the same thing. The only difference is that `lazyTransform` moves the work to the thread running `get()` while `transform` does the work in the thread that completes the future.

We now return RpcFuture instead of ListenableFuture, so the path is never taken. Additionally, the function is so small it's unlikely this path would actually make a difference.